### DISCUSSION TITLE
Universal Command Palette

### DISCUSSION THREAD

**Discussion by denolfe** [payloadcms/payload#3159] [Feature Requests & Ideas]
# Feature Request

A command palette accessible via Cmd+K. 

## Detail

This is a very common feature in many apps these days: Notion, Linear, GitHub, etc.

- Should be context-aware. Show appropriate options if on dashboard, list view, edit, etc.
- Many great libraries out there to easily implement this
  - https://github.com/pacocoursey/cmdk
  - https://github.com/albingroen/react-cmdk

### Example using [cmdk](https://cmdk.paco.me)

<img width="806" alt="CleanShot 2023-08-11 at 09 11 38" src="https://github.com/payloadcms/payload/assets/65888/41af0849-ff30-4934-98a9-5814952c3906">

**Comment by JarrodMFlesch** [comment]
I think this is a neat idea. We should gather a running list of things that ours should be able to do. I imagine it being extensible, but just wondering out of the box what we could offer!

**Comment by notchris** [comment]
This would be super cool. Would the current route change the order of the suggestions? I can imagine the desired actions changing between being on the editor / navigating a collection list.

**Reply by denolfe** [reply]
Yes, it should be "context-aware" and show relevant commands accordingly. We may have to play around with what that looks like specifically for Payload, though.

**Comment by paulpopus** [comment]
Would love this with following suggestions, just brain dumping:
- Route/collection specific commands we can extend
- Access permissions to commands
- Ability to turn off or customise core-provided commands
It should be powerful enough that we can use it with custom/plugin added routes and maybe provide some helpers to deal with multistep commands that require an additional input?

Out of the box:
- Publish with a suggestion of recently edited draft items
- Duplicate current item
- Add a new collection item -> select collection
- Preview current item (opens a live preview in the future, currently can open the preview link)
- Trigger certain emails, eg password reset, email verification
- Recently edited items
- Search (all collection items would be super useful)

**Comment by tylandavis** [comment]
This would be a great addition. There's also [this discussion on adding actions](https://github.com/payloadcms/payload/discussions/2755) which might align well with this.

We'll want to consider what might be useful for editor/marketer/content people, with intuitive ways to utilize this that aren't keyboard only.

**Comment by nathanclevenger** [comment]
I would add, that this would also be perfect to hook into app-wide search using Atlas full-text and/or vector search.  

Stacker is a great example of how to build a universal command palette in an admin app:
https://support.stackerhq.com/hc/en-us/articles/4928555280275-Shortcuts#:~:text=To%20open%20the%20command%20bar,or%20Ctrl%20%2B%20K%20on%20Windows.
